crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -418,6 +484,76 @@ impl TransactionBlockEffects {
             TransactionBlockEffectsKind::DryRun { native, .. } => native,
         }
     }
+
+    /// Get the transaction data from the transaction block effects.
+    /// Will error if the transaction data is not available/invalid, but this should not occur.
+    fn transaction_data(&self) -> Result<NativeTransactionData> {
+        Ok(match &self.kind {
+            TransactionBlockEffectsKind::Stored { stored_tx, .. } => {
+                let s: NativeSenderSignedData = bcs::from_bytes(&stored_tx.raw_transaction)
+                    .map_err(|e| {
+                        Error::Internal(format!("Error deserializing transaction data: {e}"))
+                    })?;
+                s.transaction_data().clone()
+            }
+            TransactionBlockEffectsKind::Executed { tx_data, .. } => {
+                tx_data.transaction_data().clone()
+            }
+            TransactionBlockEffectsKind::DryRun { tx_data, .. } => tx_data.clone(),
+        })
+    }
+
+    /// Get the programmable transaction from the transaction block effects.
+    /// * If the transaction was unable to be retrieved, this will return an Err.
+    /// * If the transaction was able to be retrieved but was not a programmable transaction, this
+    ///   will return Ok(None).
+    /// * If the transaction was a programmable transaction, this will return Ok(Some(tx)).
+    fn programmable_transaction(&self) -> Result<Option<ProgrammableTransaction>> {
+        let tx_data = self.transaction_data()?;
+        match tx_data.into_kind() {
+            NativeTransactionKind::ProgrammableTransaction(tx) => Ok(Some(tx)),
+            _ => Ok(None),
+        }
+    }
+
+    /// Resolves the module ID within a Move abort to the storage ID of the package that the
+    /// abort occured in.
+    /// * If the error is not a Move abort, or the Move call in the programmable transaction cannot
+    ///   be found, this function will do nothing.
+    /// * If the error is a Move abort and the storage ID is unable to be resolved an error is
+    ///   returned.
+    async fn resolve_native_status_impl(
+        &self,
+        resolver: &PackageResolver,
+        status: &mut NativeExecutionStatus,
```

Comment:
nit: can we extract this via `self` as well? I don't think it makes sense for an instance of this type to offer to resolve any status, when that resolution depends on its own PTB commands.